### ISSUE TITLE
Brew install socat not working

### ISSUE THREAD

**Issue by amberdiehl** [Homebrew/brew#2781]
I am trying to install socat and it does not appear to be installing correctly. Brew install runs without error but when I try to run socat, I get "command not found."

**[2017-06-13T09:52:49]/$ brew install socat**
==> Downloading https://homebrew.bintray.com/bottles/socat-1.7.3.2.sierra.bottle.tar.gz
Already downloaded: /Users/amberdiehl/Library/Caches/Homebrew/socat-1.7.3.2.sierra.bottle.tar.gz
==> Pouring socat-1.7.3.2.sierra.bottle.tar.gz
==> Using the sandbox
🍺  /usr/local/Cellar/socat/1.7.3.2: 9 files, 649.5KB

**/usr/local/bin**
lrwxr-xr-x    1 amberdiehl  admin        33 Jun 13 09:53 socat -> ../Cellar/socat/1.7.3.2/bin/socat

**[2017-06-13T09:57:10]/usr/local/Cellar/socat/1.7.3.2/bin$ ls -la**
total 304
drwxr-xr-x  4 amberdiehl  admin    136 Jun 13 09:53 .
drwxr-xr-x  9 amberdiehl  admin    306 Jun 13 09:53 ..
-r-xr-xr-x  1 amberdiehl  admin  77868 Jun 13 09:53 filan
-r-xr-xr-x  1 amberdiehl  admin  72728 Jun 13 09:53 procan

**Brew config and brew doctor:
[2017-06-13T09:58:35]~$ brew config**
HOMEBREW_VERSION: 1.2.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: ed9942fbd1ec4243f0a92ab8f9b2411c8b1fb091
Last commit: 24 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: fdaf5a6e13333d10aec2bedd59eab76a2f1b3938
Core tap last commit: 3 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit broadwell
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.8.4 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.8.0_102
macOS: 10.12.5-x86_64
Xcode: 8.0
CLT: 8.3.2.0.1.1492020469
X11: N/A
**[2017-06-13T09:59:24]~$ brew doctor**
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and what additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew provided
script of the same name. We found the following "config" scripts:
  /Library/Frameworks/Python.framework/Versions/3.5/bin/python3-config
  /Library/Frameworks/Python.framework/Versions/3.5/bin/python3.5-config
  /Library/Frameworks/Python.framework/Versions/3.5/bin/python3.5m-config

Warning: Python is installed at /Library/Frameworks/Python.framework

Homebrew only supports building against the System-provided Python or a
brewed Python. In particular, Pythons installed to /Library can interfere
with other software installs.

Warning: Some directories in your path end in a slash.
Directories in your path should not end in a slash. This can break other
doctor checks. The following directories should be edited:
  /Applications/MAMP/Library/bin/

Warning: /usr/bin occurs before /usr/local/bin
This means that system-provided programs will be used instead of those
provided by Homebrew. The following tools exist at both paths:
  2to3
  easy_install

Consider setting your PATH so that /usr/local/bin
occurs before /usr/bin. Here is a one-liner:
  echo 'export PATH="/usr/local/bin:$PATH"' >> ~/.bash_profile

Warning: Your Xcode (8.0) is outdated.
Please update to Xcode 8.3.3 (or delete it).
Xcode can be updated from the App Store.

Warning: Broken symlinks were found. Remove them with `brew prune`:
  /usr/local/bin/socat

Warning: Some installed formula are missing dependencies.
You should `brew install` the missing dependencies:
  brew install openssl@1.1

Run `brew missing` for more details.

Warning: Some keg-only formula are linked into the Cellar.
Linking a keg-only formula, such as gettext, into the cellar with
`brew link <formula>` will cause other formulae to detect them during
the `./configure` step. This may cause problems when compiling those
other formulae.

Binaries provided by keg-only formulae may override system binaries
with other strange results.

You may wish to `brew unlink` these brews:
  freetds@0.91
  zlib

**Comment by ilovezfs** [comment]
Maybe try
```
brew reinstall socat
```

**Comment by MikeMcQuaid** [comment]
> Warning: "config" scripts exist outside your system or Homebrew directories.
> ./configure scripts often look for *-config scripts to determine if
> software packages are installed, and what additional flags to use when
> compiling and linking.
> 
> Having additional scripts in your path can confuse software installed via
> Homebrew if the config script overrides a system or Homebrew provided
> script of the same name. We found the following "config" scripts:
> /Library/Frameworks/Python.framework/Versions/3.5/bin/python3-config
> /Library/Frameworks/Python.framework/Versions/3.5/bin/python3.5-config
> /Library/Frameworks/Python.framework/Versions/3.5/bin/python3.5m-config
> 
> Warning: Python is installed at /Library/Frameworks/Python.framework
> 
> Homebrew only supports building against the System-provided Python or a
> brewed Python. In particular, Pythons installed to /Library can interfere
> with other software installs.
> 
> Warning: Some directories in your path end in a slash.
> Directories in your path should not end in a slash. This can break other
> doctor checks. The following directories should be edited:
> /Applications/MAMP/Library/bin/
> 
> Warning: /usr/bin occurs before /usr/local/bin
> This means that system-provided programs will be used instead of those
> provided by Homebrew. The following tools exist at both paths:
> 2to3
> easy_install
> 
> Consider setting your PATH so that /usr/local/bin
> occurs before /usr/bin. Here is a one-liner:
> echo 'export PATH="/usr/local/bin:$PATH"' >> ~/.bash_profile
> 
> Warning: Your Xcode (8.0) is outdated.
> Please update to Xcode 8.3.3 (or delete it).
> Xcode can be updated from the App Store.
> 
> Warning: Broken symlinks were found. Remove them with brew prune:
> /usr/local/bin/socat
> 
> Warning: Some installed formula are missing dependencies.
> You should brew install the missing dependencies:
> brew install openssl@1.1
> 
> Run brew missing for more details.
> 
> Warning: Some keg-only formula are linked into the Cellar.
> Linking a keg-only formula, such as gettext, into the cellar with
> brew link <formula> will cause other formulae to detect them during
> the ./configure step. This may cause problems when compiling those
> other formulae.
> 
> Binaries provided by keg-only formulae may override system binaries
> with other strange results.
> 
> You may wish to brew unlink these brews:
> freetds@0.91
> zlib

Please try to fix these.

**Comment by amberdiehl** [comment]
Hi there!

I will do my best to address these -- I am curious if there are some, in your opinion, that are more likely to be contributing to this issue? Needing to address Python is confusing to me--version 2 was preinstalled on my Mac and I've been installing 3 using brew.

Also, it just so happens that I went ahead and followed the advice to **reinstall** socat. By accident, I ran the command and then flipped over to finder while it was running. Strangely, for just a moment, I see socat in the directory with filan and procan and then it disappears. Does that point to something more specific that I can do to fix this?

Thanks in advance.
